pandas/tests/frame/methods/test_between_time.py

```diff
@@ -207,3 +208,106 @@ def test_between_time_datetimeindex(self):
         tm.assert_frame_equal(result, expected)
         tm.assert_frame_equal(result, expected2)
         assert len(result) == 12
+
+    # GH40245
+    @pytest.mark.parametrize("inc_start", (True, False))
+    @pytest.mark.parametrize("inc_end", (True, False))
+    def test_between_time_warn(self, inc_start, inc_end, frame_or_series):
+        rng = date_range("1/1/2000", "1/5/2000", freq="5min")
+        ts = DataFrame(np.random.randn(len(rng), 2), index=rng)
+        if frame_or_series is not DataFrame:
+            ts = ts[0]
+
+        stime = time(0, 0)
+        etime = time(1, 0)
+
+        match = (
+            "`include_start` and `include_end` "
+            "are deprecated infavour of `inclusive`."
+        )
+        with tm.assert_produces_warning(FutureWarning, match=match):
+            filtered = ts.between_time(stime, etime, inc_start, inc_end)
+
+        # new new new
+        exp_len = 13 * 4 + 1
+        if not inc_start:
+            exp_len -= 5
+        if not inc_end:
+            exp_len -= 4
+
+        assert len(filtered) == exp_len
+        # for rs in filtered.index:
+        #     t = rs.time()
+        #     if inc_start:
+        #         assert t >= stime
+        #     else:
+        #         assert t > stime
+
+        #     if inc_end:
+        #         assert t <= etime
+        #     else:
+        #         assert t < etime
+
+        # result = ts.between_time("00:00", "01:00")
+        # expected = ts.between_time(stime, etime)
+        # tm.assert_equal(result, expected)
+
+        # # across midnight
+        # rng = date_range("1/1/2000", "1/5/2000", freq="5min")
+        # ts = DataFrame(np.random.randn(len(rng), 2), index=rng)
+        # if frame_or_series is not DataFrame:
+        #     ts = ts[0]
+        # stime = time(22, 0)
+        # etime = time(9, 0)
+
+        # with tm.assert_produces_warning(FutureWarning):
+        #     filtered = ts.between_time(stime, etime, inc_start, inc_end)
+        # exp_len = (12 * 11 + 1) * 4 + 1
+        # if not inc_start:
+        #     exp_len -= 4
+        # if not inc_end:
+        #     exp_len -= 4
+
+        # assert len(filtered) == exp_len
+        # for rs in filtered.index:
+        #     t = rs.time()
+        #     if inc_start:
+        #         assert (t >= stime) or (t <= etime)
+        #     else:
+        #         assert (t > stime) or (t <= etime)
+
+        #     if inc_end:
+        #         assert (t <= etime) or (t >= stime)
+        #     else:
+        #         assert (t < etime) or (t >= stime)
+
+    # GH40245
+    def test_between_time_incorr_arg_inclusive(self):
+        rng = date_range("1/1/2000", "1/5/2000", freq="5min")
+        ts = DataFrame(np.random.randn(len(rng), 2), index=rng)
+
+        stime = time(0, 0)
+        etime = time(1, 0)
+        inclusive = "neeither"
+        msg = (
+            "Inclusive has to be either string of 'both','left', 'right', "
+            "or 'neither'. Got neeither."
```

Comment:
```suggestion
        inclusive = "bad_string"
        msg = (
            "Inclusive has to be either string of 'both','left', 'right', "
            "or 'neither'. Got bad_string."
```
I know this is tempting to reflect likely user patterns but always best to have clearer tests